### PR TITLE
Documentation setup

### LINKED ISSUE #46: Documentation
To support the usability and extensibility of photo-decomp as a stand-alone plugin, we need to develop documentation. The documentation should provide a complete API reference, answer common FAQs, and show common use cases with examples. Further, the documentation should provide a high level overview of the entire system and crucial parts so that new users can understand how to improve the system.

From a development perspective, certain components will need to have the design fully described both pictorially and in written language for posterity. 

# External, User documentation
This is the documentation for potential users.
- We'd like it to look like the documentation for [xarray](https://docs.xarray.dev/en/latest/contributing.html#contributing-to-the-documentation) and [matplotlib](https://matplotlib.org/stable/index.html) because this style of documentation is common and very clean looking

# Internal, Developer documentation
- This should be kept in a filesystem internal to NCAR employees to keep track of detailed technical designs.

### PR DESCRIPTION
**Pull request by K20shores**
This should close #46 and #47. The instructions for building and the documentation are in #47. What I didn't do was add the [sphinx fortran extension](https://github.com/VACUMM/sphinx-fortran).
Instructions:

```
conda create --name ncar
conda activate ncar
conda install sphinx
conda install sphinx-book-theme
```

- At [https://ncar.github.io/photo-decomp/switcher.json](https://ncar.github.io/photo-decomp/switcher.json) is the file that controls the version picker for *each* version of the documentation. The version picker is built dynamically.
- The picker's name is determined by the `release` version which is set in `doc/source/conf.py`
- So, the process for building documentation that is properly versioned and has links to all other versions is this:
  - Develop the new code and once done, switch to the `gh-pages` branch and merge your code in. Continue in `gh-pages` to update the documentation
  - Update the documentation as needed
  - Test the documentation by running `make html` from within the `doc/` directory and viewing the file located at `doc/build/html/index.html` in a browser on your computer
  - Once satisfied with the documentation, you must build the documentation twice
    - open up `docs/switcher.json`
      - the first object of the json file points to the stable version. Update the name of that version
      - also, add another object for the new version. This means that there will be two objects that point to current version, one called `v<whatever number> (stable)` and another called `v<whatever number>`
    - in `doc/source/conf.py`, set the `release` version variable to `v<whatever number>` and from the `doc` directory, run `make github`
      - go to the `docs/version` directory and rename the `stable` directory to whatever the current version is
    - in `doc/source/conf.py`, set the `release` version variable to `stable` and from the `doc` directory, run `make github`
      - this will create the `docs/verions/stable` directory to contain the most recent version
    - You will now have updated the `docs/stable` documentation and created a new set of documentation at `docs/v<whatever number>`. This is ready to be pushed to github, after a few minutes, the new version will be live at [https://ncar.github.io/photo-decomp/versions/stable](https://ncar.github.io/photo-decomp/versions/stable)

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Photo-decomp'
+copyright = '2022, NCAR/UCAR'
+author = 'NCAR/UCAR'
+
+# The full version, including alpha/beta/rc tags
+release = '1.0'
```
Comment (mattldawson) [inline doc/source/conf.py:25]: Should we start at v0.1? We should come up with a plan for how to version this tool.

File: docs/switcher.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[
+    {
+        "name": "v1.0 (stable)",
+        "version": "stable",
+        "url": "https://ncar.github.io/photo-decomp/versions/stable"
+    },
+    {
+        "name": "v1.0",
+        "version": "1.0",
+        "url": "https://ncar.github.io/photo-decomp/versions/1.0"
```
Comment (mattldawson) [inline docs/switcher.json:10]: ... maybe v0.1 here also

**Review by mattldawson — APPROVED** [review]
Looks good to me! Just a couple minor comments.

One question - is there a way to merge the `doc/` and `docs/` folders? (We could also do this as part of #76 as this will probably involve some changes to where/how the documentation is built.)

File: docs/switcher.json (hunk 2 above)
Comment (K20shores) [reply docs/switcher.json:10]: sure, I'll make that change and merge /doc and /docs if I can

**Comment by K20shores** [comment]
For the future, maybe we should think about placing the version in its own file so that both the python process and the Makefile can read it and remove that manual step for creating both the stable version of the docs and the new one

**Comment by mattldawson** [comment]
I think once we do #76 that this will remove a lot of the manual steps. We could probably have a release branch that we only push to once we're ready to release a new version. Pushing to the branch (or maybe releasing a new version - not sure if this is possible) could trigger the action that handles some of these steps

**Pull request merged**